pkg/setting/setting.go

```diff
@@ -860,10 +860,34 @@ func (cfg *Cfg) Load(args *CommandLineArgs) error {
 	if err != nil {
 		return err
 	}
-	cfg.LoginMaxInactiveLifetimeDays = auth.Key("login_maximum_inactive_lifetime_days").MustInt(7)
-
-	LoginMaxLifetimeDays = auth.Key("login_maximum_lifetime_days").MustInt(30)
-	cfg.LoginMaxLifetimeDays = LoginMaxLifetimeDays
+	maxInactiveDaysVal := auth.Key("login_maximum_inactive_lifetime_days").MustInt(7)
+	maxInactiveDurationVal, err := valueAsString(auth, "login_maximum_inactive_lifetime_duration", "168h")
+	if err != nil {
+		return err
+	}
+	if maxInactiveDaysVal != 7 {
+		cfg.LoginMaxInactiveLifetimeDuration = time.Duration(maxInactiveDaysVal) * 24 * time.Hour
+	} else {
+		cfg.LoginMaxInactiveLifetimeDuration, err = time.ParseDuration(maxInactiveDurationVal)
+		if err != nil {
+			return err
+		}
+	}
+	maxLifetimeDaysVal := auth.Key("login_maximum_lifetime_days").MustInt(30)
+	maxLifetimeDurationVal, err := valueAsString(auth, "login_maximum_lifetime_duration", "720h")
+	if err != nil {
+		return err
+	}
+	if maxLifetimeDaysVal != 30 {
+		cfg.LoginMaxLifetimeDuration = time.Duration(maxLifetimeDaysVal) * 24 * time.Hour
+		LoginMaxLifetimeDays = float64(maxLifetimeDaysVal * 24)
+	} else {
+		cfg.LoginMaxLifetimeDuration, err = time.ParseDuration(maxLifetimeDurationVal)
+		LoginMaxLifetimeDays = cfg.LoginMaxLifetimeDuration.Hours()
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Suggest to simplify this by treating old settings as deprecated. Pseudo code here, but you probably need to refactor it a bit to get proper error handling. Need to import gtime package at the top as well I think.

```suggestion
	maxInactiveDaysVal := auth.Key("login_maximum_inactive_lifetime_days").MustString("")
	if maxInactiveDaysVal != "" {
		maxInactiveDaysVal = fmt.Sprintf("%sd", maxInactiveDaysVal)
		cfg.Logger.Warn("[Deprecated] the configuration setting 'login_maximum_inactive_lifetime_days' is deprecated, please use 'login_maximum_inactive_lifetime_duration' instead")
	} else {
		maxInactiveDaysVal = "7d"
	}
	
	maxInactiveDurationVal, err = valueAsString(auth, "login_maximum_inactive_lifetime_duration", maxInactiveDaysVal)
	if err != nil {
		return err
	}
	
	cfg.LoginMaxInactiveLifetime, err = gtime.ParseInterval(maxInactiveDurationVal)
	if err != nil {
		return err
	}
	
	maxLifetimeDaysVal := auth.Key("login_maximum_lifetime_days").MustString("")
	if maxLifetimeDaysVal != "" {
		maxLifetimeDaysVal = fmt.Sprintf("%sd", maxLifetimeDaysVal)
		cfg.Logger.Warn("[Deprecated] the configuration setting 'login_maximum_lifetime_days' is deprecated, please use 'login_maximum_lifetime_duration' instead")
	} else {
		maxLifetimeDaysVal = "7d"
	}
	
	maxLifetimeDurationVal, err = valueAsString(auth, "login_maximum_lifetime_duration", maxLifetimeDaysVal)
	if err != nil {
		return err
	}
	
	cfg.LoginMaxLifetime, err = gtime.ParseInterval(maxLifetimeDurationVal)
	if err != nil {
		return err
	}
	LoginMaxLifetime = cfg.LoginMaxLifetime
```